aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -34,7 +34,7 @@ DEFINE_DISPATCH(max_values_stub);
 
 static inline Tensor integer_upcast(const Tensor& self, optional<ScalarType> dtype) {
   ScalarType scalarType = self.scalar_type();
-  ScalarType upcast_scalarType = dtype.value_or(at::isIntegralType(scalarType) ? ScalarType::Long : scalarType);
+  ScalarType upcast_scalarType = dtype.value_or(at::isIntegralType(scalarType, /*includeBool*/ false) ? ScalarType::Long : scalarType);
```

Comment:
err...won't this break sum(bool) -> int64?

Reply:
sum(bool) returns either tensor(True) or tensor(False) 
torch.sum(bool) works either way

Reply:
will break cumsum/cumprod.
Due to https://github.com/pytorch/pytorch/issues/20376 this wasn't caught. Will make sure that its fixes asap

Reply:
ok, so are you going to change this to true?